go/filemanager/pfsmodules/chunkmeta.go

```diff
@@ -0,0 +1,203 @@
+package pfsmodules
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"sort"
+	"strconv"
+)
+
+const (
+	defaultMaxChunkSize = 4 * 1024 * 1024
+	defaultMinChunkSize = 4 * 1024
+)
+const (
+	ChunkMetaCmdName = "GetChunkMeta"
+)
+
+// ChunkMeta holds the chunk meta's info.
+type ChunkMeta struct {
+	Offset   int64  `json:"offset"`
+	Checksum string `json:"checksum"`
+	Len      int64  `json:"len"`
+}
+
+// ChunkMetaCmd is a command.
+type ChunkMetaCmd struct {
+	Method    string `json:"method"`
+	FilePath  string `json:"path"`
+	ChunkSize int64  `json:"chunksize"`
+}
+
+// ToURLParam encodes ChunkMetaCmd to URL encoding string.
+func (p *ChunkMetaCmd) ToURLParam() url.Values {
```

Comment:
These code are repetitive, a good candidate for automatically generating them. See: https://blog.golang.org/generate

Not required, just want to let you know :)